From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Fri, 10 Jun 2022 10:28:17 +0800
Subject: [PATCH] x86/kexec: add module header file to kexec.h
Patch-mainline: Never, it is needed because of patches.openEuler/1669
Modified-by-SEL: N/A
References: bsn#19

After sync patches from v5.10.121 stable, we see below errors.

In file included from kernel/crash_core.c:15:0:
./arch/x86/include/asm/kexec.h:213:10: error: unknown type name ‘Elf_Shdr’; did you mean ‘Elf64_Shdr’?
          Elf_Shdr *section,
          ^~~~~~~~
          Elf64_Shdr
./arch/x86/include/asm/kexec.h:214:16: error: unknown type name ‘Elf_Shdr’
          const Elf_Shdr *relsec,
                ^~~~~~~~
./arch/x86/include/asm/kexec.h:215:16: error: unknown type name ‘Elf_Shdr’
          const Elf_Shdr *symtab);
                ^~~~~~~~
make[1]: *** [scripts/Makefile.build:281: kernel/crash_core.o] Error 1

It is casused by

patches.openEuler/1669-x86-kdump-move-reserve_crashkernel-_low-into-crash_c.patch

which added this hunk in kernel/crash_core.c

+#ifdef CONFIG_KEXEC_CORE
+#include <asm/kexec.h>
+#endif

To resovle compile issue, the module.h is needed.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/x86/include/asm/kexec.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/include/asm/kexec.h b/arch/x86/include/asm/kexec.h
index f4b011f11336..e4c94299750a 100644
--- a/arch/x86/include/asm/kexec.h
+++ b/arch/x86/include/asm/kexec.h
@@ -46,6 +46,7 @@
 
 #include <linux/string.h>
 #include <linux/kernel.h>
+#include <linux/module.h>
 
 #include <asm/page.h>
 #include <asm/ptrace.h>
-- 
2.34.1

